api/v1alpha1/release_types_test.go

```diff
@@ -0,0 +1,671 @@
+						},
+					},
+				},
+			}
+			Expect(r.IsDone()).To(BeTrue(),
+				"Expected to get: %+v ; Actually Got: %+v\n", true, false)
+		})
+
+		It("should return false because condition.Status is ConditionUnknown", func() {
+			r := &Release{
+				Status: ReleaseStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:   releaseConditionType,
+							Status: metav1.ConditionUnknown,
+						},
+					},
+				},
+			}
+			Expect(r.IsDone()).To(BeFalse(),
+				"Expected to get: %+v ; Actually Got: %+v\n", false, true)
+		})
+	})
+
+	Context("MarkFailed()", func() {
+		It("should NOOP because IsDone() is true, and Status.CompletionTime is not nil", func() {
+			args := commonArgs{
+				reason:  ReleaseReasonFoo,
+				message: "cow say m00",
+			}
+			r := &Release{
+				Status: ReleaseStatus{
+					StartTime: &metav1.Time{
+						Time: time.Now().Add(-time.Hour * 1),
+					},
+					CompletionTime: &metav1.Time{ // not nil
+						Time: time.Now(),
+					},
+					Conditions: []metav1.Condition{
+						{
+							Type:    releaseConditionType,
+							Status:  metav1.ConditionTrue,
+							Reason:  ReleaseReasonBar.String(),
+							Message: "Testcase one message string",
+						},
+					},
+				},
+			}
+			r.MarkFailed(args.reason, args.message)
+			Expect(len(r.Status.Conditions)).To(
+				Equal(1),
+				"Expected number of []Conditions to equal \"%+v\"\n",
+				1)
+			Expect(r.Status.Conditions[0]).To(MatchFields(IgnoreMissing|IgnoreExtras, Fields{
+				"Status":  Equal(metav1.ConditionTrue),
+				"Type":    Equal(releaseConditionType),
+				"Reason":  Equal(ReleaseReasonBar.String()),
+				"Message": Equal("Testcase one message string"),
+			}), "Expect condition struct fields to match expected fields")
+		})
+
+		It("should not NOOP because CompletionTime is nil, and IsDone returns false", func() {
+			args := commonArgs{
+				reason:  ReleaseReasonBar,
+				message: "what does the fox say",
+			}
+			r := &Release{
+				Status: ReleaseStatus{
+					StartTime: &metav1.Time{
+						Time: time.Now().Add(-time.Hour * 1),
+					},
+					// CompletionTime == nil
+					Conditions: []metav1.Condition{
+						{
+							Type:    releaseConditionType,
+							Status:  metav1.ConditionUnknown, // IsDone() false
+							Reason:  "Testcase two reason string",
+							Message: "Testcase two message string",
+						},
+					},
+				},
+			}
+			r.MarkFailed(args.reason, args.message)
+			Expect(len(r.Status.Conditions)).To(
+				Equal(1),
+				"Expected number of []Conditions to equal \"%+v\"\n",
+				1)
+			Expect(r.Status.Conditions[0]).To(MatchFields(IgnoreMissing|IgnoreExtras, Fields{
+				"Status":  Equal(metav1.ConditionFalse),
+				"Type":    Equal(releaseConditionType),
+				"Reason":  Equal(ReleaseReasonBar.String()),
+				"Message": Equal("what does the fox say"),
+			}), "Expect condition struct fields to match expected fields")
+		})
+	})
+
+	Context("MarkInvalid()", func() {
+		It("should not NOOP because IsDone() returns false", func() {
```

Comment:
Not really understanding the description. As a side note, better than putting `because IsDone() returns false` I would put `because is not done`.